### PR TITLE
NRPT-707: Adding emli integration.

### PR DESCRIPTION
**Pull request by marklise**
Needs new ENV var: NRIS_EMLI_API_ENDPOINT to point to emprInspections (defaults to prod api @ https://api.nrs.gov.bc.ca/nrisws-api/v1/emprInspections)

### CODE REVIEW

File: api/src/integrations/nris-emli/datasource.js [hunk 1]
```diff
@@ -0,0 +1,466 @@
...
+    newRecord._sourceRefNrisId = record.assessmentId;
+    try {
+      newRecord.dateIssued = record.inspection.inspctReportSentDate
+        ? moment.tz(record.inspection.inspctReportSentDate, 'America/Vancouver').toDate()
+        : null;
+    } catch (error) {
+      defaultLog.debug(error);
+      newRecord.dateIssued = null;
+    }
+    newRecord.issuingAgency = 'Ministry of Energy, Mines and Low Carbon Innovation';
```
Comment (BcGovNeal) [inline api/src/integrations/nris-emli/datasource.js:206]: `issuingAgency` and maybe `author` should be `EMLI` to be consistent with existing DB values.  More details in https://bcmines.atlassian.net/browse/NRPT-361

**Review by BcGovNeal — APPROVED** [review]
Other than the issuing agency value, it looks good.  Might need to address the Sonar scan results

File: api/src/integrations/nris-emli/datasource.js (hunk 1 above)
Comment (marklise) [reply api/src/integrations/nris-emli/datasource.js]: Fixed in 8fa9bd8

File: api/src/integrations/nris-emli/datasource.js [hunk 2]
```diff
@@ -0,0 +1,466 @@
...
+
+    if (record.client && record.client.length > 0 && record.client[0]) {
+      newRecord.issuedTo = {
+        type: 'Company',
+        companyName: record.client[0].orgName || '',
+        fullName: record.client[0].orgName || ''
+      };
+    }
+
+    newRecord.recordName = record.assessmentId;
```
Comment (mtCarto) [inline api/src/integrations/nris-emli/datasource.js:235]: Record name should be a bit more descriptive similar to FLNRO Inspections, was overlooked in the transformation planning table.
Eg. `EMLI Inspection - Record xxxyyy`

**Review by mtCarto — CHANGES_REQUESTED** [review]

File: api/src/integrations/nris-emli/datasource.js [hunk 3]
```diff
@@ -0,0 +1,466 @@
...
+
+      if (!payload) {
+        defaultLog.error('Error authenticating against NRIS API.');
+        return { status: 'Auth Error' };
+      }
+
+      this.token = payload.access_token;
+      defaultLog.info('NRIS API token expires:', (payload.expires_in / 60 / 60).toFixed(2), ' hours');
+
+      // Hardcoded to start in October 1, 2017.
```
[the remark this answers is not in the document]
Comment (marklise) [reply api/src/integrations/nris-emli/datasource.js]: Fixed in 0bbfd9a

File: api/src/integrations/nris-emli/datasource.js (hunk 2 above)
Comment (marklise) [reply api/src/integrations/nris-emli/datasource.js]: Fixed in 827fb0d

**Review by mtCarto — APPROVED** [review]

**Pull request merged**